scripts/ci/guardrails/check_global_state_assertions.py

```
@@ -0,0 +1,242 @@
+            continue
+        violations.append((node.lineno, message))
+    return sorted(set(violations))
+
+
+def _changed_lines_by_file() -> dict[str, set[int]]:
+    """Map file -> line numbers added/modified in the staged diff."""
+    import re
+    import subprocess
+
+    result = subprocess.run(
+        ["git", "diff", "--cached", "-U0", "--no-color", "--", "tests/"],
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)